src/transformStateWithClones.js

```diff
@@ -7,7 +7,40 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const previousStateVers = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        copyState = Object.assign(copyState, action.extraData);
+        previousStateVers.push({ ...copyState });
```

Comment:
Move the `previousStateVers.push({ ...copyState })` to the end of the loop to avoid duplicating it in each case.